src/components/visualizations/chart/highcharts/plugins/zeroAlignYAxis.ts

```diff
@@ -155,32 +127,69 @@ function minmaxCanon(minmax: IMinMaxInfo[]): ICanon[] {
     return canon;
 }
 
+function getMinMaxLookup(minmax: IMinMaxInfo[]): IMinMaxLookup {
+    return minmax.reduce((result: IMinMaxLookup, item: IMinMaxInfo) => {
+        result[item.id] = item;
+        return result;
+    }, {});
+}
+
+function calculateMin(idx: number, minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
+    const fraction = !minmax[idx].max ?
+                        minmax[idx].min : // handle divide zero case
+                        minmax[idx].min / minmax[idx].max;
+    return fraction * minmaxLookup[axisIndex].max;
+}
+
+function calculateMax(idx: number, minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
+    const fraction = !minmax[idx].min ?
+                    minmax[idx].max : // handle divide zero case
+                    minmax[idx].max / minmax[idx].min;
+    return fraction * minmaxLookup[axisIndex].min;
+}
+
 /**
- * New min is calculated base on axis having smallest min
+ * New min is calculated base on axis having smallest min in case having min/max setting
+ * Otherwise, it is calculated base on axis having smaller min
  * @param minmax
  * @param minmaxLookup
  * @param axisIndex
  */
-function getNewMin(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
-    const idx = (minmax[0].min <= minmax[1].min) ? 0 : 1;
-    const f = !minmax[idx].max ?
-                    minmax[idx].min : // handle divide zero case
-                    minmax[idx].min / minmax[idx].max;
-    return f * minmaxLookup[axisIndex].max;
+function getMin(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number) {
```

Comment:
is not possible to rewrite this two functions into one and different parts provide as parameters? Something like this. I am curious about readability and length of such solution

function getLimit(minmax: IMinMaxInfo[], minmaxLookup: IMinMaxLookup, axisIndex: number), getIndex: function, calculateLimit: function, findExtreme: function){
const isMinMaxConfig = isMinMaxConfigOnAnyAxis(minmax);
if (isMinMaxConfig) {
const idx = getIndex(minmax);
return calculateLimit(idx, minmax, minmaxLookup, axisIndex);
}
return findExtreme([0, 1]
.map((index: number) => calculateLimit(index, minmax, minmaxLookup, axisIndex)))